sklearn/utils/estimator_checks.py

```diff
@@ -1057,17 +1072,24 @@ def check_array_api_input(
     When check_values is True, it also checks that calling the estimator on the
     array_api Array gives the same results as ndarrays.
 
-    When sample_weight is True, dummy sample weights are passed to the fit call.
+    When check_sample_weight is True, dummy sample weights are passed to the
+    fit call.
+
+    When expect_only_array_outputs is False, the check accepts non-array
+    outputs from estimator methods (e.g., sparse data structures). This is
```

Comment:
Could you expand on this? I was confused just reading this, but https://github.com/scikit-learn/scikit-learn/pull/32846/changes#r2602981393 cleared it up for me.

Reply:
Thanks! Feel free to merge when you're ready!